app/controllers/events_controller.rb

```diff
@@ -1,68 +1,87 @@
-  # end
+  # before_action :authorize_request
 
-  def show; end
+  def index
+    events = retrieve_events
+    sidebar_events = create_sidebar_events(events)
+
+    render json: sidebar_events, status: 200
+  rescue ActiveRecord::RecordInvalid
+    render json: { error: 'neither event_id nor bridge_id were valid' }, status: 400 # Bad Request
+  rescue ActiveRecord::RecordNotFound
+    render json: { error: 'events matching that id were not found' }, status: 400
+  rescue ActiveRecord::NotNullViolation
+    render json: { error: 'neither event_id nor bridge_id were not submitted' }, status: 400 # Bad Request
+  end
+
+  def show
+    event = Event.find(event_params[:id])
+    render json: event, status: 200
+  rescue ActiveRecord::RecordNotFound
+    render json: { error: 'an event by that id was not found' }, status: 400
+  end
+
+  def destroy
+    event = Event.find(event_params[:id])
+    event.destroy
+  end
 
-  # receive bridge id + data
   def create
-    bridge = Bridge.find(params[:id])
-    data = { inbound: request.body, outbounds: [] }
-    event = Event.new(data: data, bridge_id: bridge.id)
-    if event.save
-      # EventWorker.perform(id of event just created)
-      status 201 # Created
-    else
-      status 400 # Bad Request
-    end
+    event = create_event_object(create_data_object, find_bridge)
+    event.save!
+    EventWorker.perform_async(event.id)
+    render json: {}, status: 202 # Accepted (asynchronous processing)
+  rescue ActiveRecord::RecordNotFound
+    render json: { error: 'a bridge by that id was not found' }, status: 400
+  rescue ActiveRecord::RecordInvalid
+    render json: { error: 'payload, bridge_id, or urls were invalid' }, status: 400 # Bad Request
+  rescue ActiveRecord::NotNullViolation
+    render json: { error: 'payload, bridge_id, or urls fields were not submitted' }, status: 400 # Bad Request
   end
 
   private
 
-  def date_format(_date)
-    year = time.split('-')[0]
-    month = time.split('-')[1]
-    day = time.split('-')[2]
-    "#{year}-#{month}-#{day}"
+  def event_params
+    params.permit(:id, :bridge_id, :event_id, :test)
   end
 
-  def set_user
-    # bridge_id or #event_id
-    # => User
+  def create_data_object
+    datetime = DateTime.now.utc.to_s.split(' ')
+    { 'inbound' => {
+      'payload' => JSON.parse(request.body.read),
+      'date' => datetime.first,
+      'time' => datetime[1],
+      'ip' => request.ip,
+      'content_length' => request.content_length
+    },
+      'outbound' => [] }
   end
-end
 
-# Step 1
-# Convert binary to jsonb
+  def create_event_object(data, bridge)
+    Event.new(
+      data: data.to_json,
+      bridge_id: bridge.id,
+      test: event_params[:test] || false
+    )
+  end
 
-# Step 2
-# payload:
-{
-  test: 'user entered string from editor',
-  production: 'user entered string from editor'
-}
+  def create_sidebar_events(events)
+    events.map(&:sidebar_format).sort_by { |event| event[:id] }.reverse
```

Comment:
This would be really slow. Instead of iterating, why not push this kind of this into the db? 

```ruby
Event.find(event_params[:event_id]).bridge.events.limit(100).order(completed_at: :desc)
```

That would make a db query *similar* to:

```sql
SELECT * FROM events
WHERE bridge_id = x
LIMIT 100
ORDER_BY completed_at DESC
```

Reply:
The main point of that method is to format each event object to the format of the sidebar, which requires iteration. 
If I add your `limit(100).order(completed_at: :desc)` to the events being passed in, that will speed it up some by alleviating the `sort_by` and `reverse` calls here.

Reply:
Yeah lets format it on the frontend. You can just send me the raw timestamp. I already have to iterate through events. No reason to add another iteration before.

Reply:
I can see I've wasted a lot of time massaging data for easy front-end consumption.  Next time I shouldn't look at the design or UI when writing controllers, but just a list of data needs for each action/endpoint.